data/patches/miq_application.js.diff

```diff
@@ -35,5 +35,60 @@
 +// --------------
 +
 +
++// setTimeout and clearTimeout monitoring
++// ---------------
++
++// Set up container to hold timeout and interval state
++window.testing = {
++  timeouts: {}
++};
++
++// Make a copy of the original setTimeout function
++window._setTimeout = window.setTimeout;
++window.setTimeout = function(callback, timeout) {
++  // We need a handle to store our timeout under, we can't just use the
++  // timeout ID because we don't know it until after we create the timeout
++  var handle = _.uniqueId();
```

Comment:
Ah, I see it now... we don't care about the handle, it's used as a key. We care about the IDs returned by setTimeout... got it.